src/components/scenes/CoinRankingScene.tsx

```diff
@@ -194,30 +212,35 @@ const CoinRankingComponent = (props: Props) => {
 
   // Subscribe to changes to the current data set:
   React.useEffect(() => {
-    let abort = () => {}
+    let loopAbort = () => {}
     // Refresh from the beginning periodically
     let timeoutId = setTimeout(loopBody, LISTINGS_REFRESH_INTERVAL)
     function loopBody() {
       debugLog(LOG_COINRANK, 'Refreshing list')
-      const abortable = queryLoop(1)
-      abort = abortable.abort
-      abortable.promise
+      const abortableQueryLoop = queryLoop(1)
+      loopAbort = abortableQueryLoop.abort
+      abortableQueryLoop.promise
         .catch(e => console.error(`Error in query loop: ${e.message}`))
         .finally(() => {
           timeoutId = setTimeout(loopBody, LISTINGS_REFRESH_INTERVAL)
         })
     }
 
     return () => {
-      // Reset related query state when this effect is unmounted:
-      clearTimeout(timeoutId)
-      abort()
-      pageQueryAbortRef.current()
-      coinRanking.coinRankingDatas = []
-      lastStartIndex.current = 1
-      setDataSize(0)
-      setRequestDataSize(QUERY_PAGE_SIZE)
+      // Reset related query state when this effect is unmounted (coingeckoFiat
+      // changed), but the scene is still mounted:
+      if (mounted.current) {
+        debugLog(LOG_COINRANK, `Clearing coinRankingDatas cache for new fiat: ${coingeckoFiat}`)
+        clearTimeout(timeoutId)
+        loopAbort()
+        pageQueryAbortRef.current()
+        coinRanking.coinRankingDatas = []
+        lastStartIndex.current = 1
+        setDataSize(0)
+        setRequestDataSize(QUERY_PAGE_SIZE)
+      }
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why was this `// eslint-disable-next-line react-hooks/exhaustive-deps` added? Is there an alternative?

Reply:
This was the cause of the actual bug in the task - this cleanup fn is appropriate only for when the scene is still mounted, implying that the fiat changed while the scene was mounted, so clearing everything was appropriate.

It's inappropriate to run this cleanup when the scene unmounts, which was happening in the original implementation.

Reply:
What's the reason for the `eslint-disable-next-line` rule hack though? Seeing this hack is a red flag. I'm trying to understand why it's necessary and perhaps why another approach wasn't taken.